### DISCUSSION TITLE
Tasmota can no longer be compiled in version 14.3 for the ESP32 C2.

### DISCUSSION THREAD

**Discussion by Basti149** [arendst/Tasmota#22317] [Support request] [answered]
I have a problem when I download the zip file from Github and then unzip it. To then compile it myself with the help of Visual Studio Code. I get the following error message when creating the build TasmotaESP32C2 (see picture). I have not made any changes, it works with Tasmota32

![grafik](https://github.com/user-attachments/assets/c4e469ca-e53f-4ce7-9938-e593f15561c0)

**Comment by Jason2866** [comment]
See https://github.com/arendst/Tasmota/discussions/21941

**Reply by Basti149** [reply]
But it still worked with the Tasmota version v13.x. I still have the version VS 1.70.3

**Reply by Jason2866** [reply, marked as answer]
and? A lot added and changed since that time. Too much for Windows. Install and use WSL to solve.

**Reply by Basti149** [reply]
I first have to create a Vm with Windows, right?

**Reply by Wintespe** [reply]
A good alternative to WSL is to install Linux MInt in a virtual machine with VirtualBox. There are then no side effects... in the Windows installation. I was able to compile Tasmota there without any problems #22068.

**Comment by Wintespe** [comment]
Try to change in platformio_override.ini

```
[platformio]
core_dir = .platformio
```

This works in my case.

**Reply by Basti149** [reply]
So here??

![grafik](https://github.com/user-attachments/assets/a27081bb-4c93-43b2-be96-ec2ef530cf4d)

**Comment by Wintespe** [comment]
Yes

**Reply by Wintespe** [reply]
To test again, I

> reinstalled Windows 10 (fresh machine with latest updates)
> installed VisualStudioCode and PlatformIO
> downloaded Tasmota 14.3.0 source code

Set **env:tasmota32c2(Tasmota-14.3.0)**

Activate core_dir = **.platformio** and compile

```
[platformio]
; For best Gitpod performance remove the ";" in the next line. Needed Platformio files are cached and installed at first run
core_dir = .platformio
; For unrelated compile errors with Windows it can help to shorten Platformio project path
;workspace_dir = c:\.pio
```

```
14.3.0\.pio\build\tasmota32c2\firmware.map"
Total sizes:
Used stat D/IRAM:   97472 bytes (  93872 remain, 50.9% used)
      .data size:   11480 bytes
      .bss  size:   48088 bytes
      .text size:   37904 bytes
Used Flash size : 1861286 bytes
           .text: 1241094 bytes
         .rodata:  619936 bytes
Total image size: 1910670 bytes (.bin may be padded larger)
======================================================================== [SUCCESS] Took 2404.27 seconds ========================================================================
Environment    Status    Duration
-------------  --------  ------------
tasmota32c2    SUCCESS   00:40:04.272

```

Now comment out core_dir = .**platformio** and compile

```
[platformio]
; For best Gitpod performance remove the ";" in the next line. Needed Platformio files are cached and installed at first run
;core_dir = .platformio
; For unrelated compile errors with Windows it can help to shorten Platformio project path
;workspace_dir = c:\.pio
```

**Failed to compile**

```
Compiling .pio\build\tasmota32c2\lib22a\FS\FS.cpp.o
riscv32-esp-elf-g++: fatal error: cannot execute 'C:/Users/Win10/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/13.2.0/../../../../riscv32-esp-elf/bin/as.exe': CreateProcess: No such file or directory
compilation terminated.
*** [.pio\build\tasmota32c2\src\tasmota.ino.cpp.o] Error 1
```

Unfortunately, I have no explanation why **core_dir = .platformio** fixes the bug. But I hope the Tasmota team can do something with the result.

**Reply by Jason2866** [reply]
Yes, it shorten the path of the resulting cmd for compile and link and assemble run.

**Reply by Wintespe** [reply]
I changed
**core_dir = .platformio**
and not
**workspace_dir = c:\.pio**

why does **core_dir = .platformio** shorten the path?
If so, the comments about it are strange

**Reply by Jason2866** [reply]
Because the Platformio packages are in the project and not "elsewhere".  The comments are not strange. If build online with Gitpod the Platformio packages are not installed every time when the Workspace is started.

**Reply by Wintespe** [reply]
Thanks for the answer. But my question was about **shorten the path**.

I do not build online with Gitpod, the build environment is local. What does the **core_dir = .platformio** statement do in this case? This may be the key to the problem.

**Reply by Jason2866** [reply]
Platformio documentation https://docs.platformio.org/en/latest/envvars.html

**Reply by Wintespe** [reply]
From
 https://docs.platformio.org/en/latest/envvars.html

```
Directories
PLATFORMIO_CORE_DIR
Allows one to override “platformio.ini” (Project Configuration File) option core_dir.

It may need to re-install PlatformIO Core (CLI) (remove default core directory) to take effect.
```
This do not help to find the root cause, thanks anyway.
